packages/sdk/src/types/Vouchers.ts

```diff
@@ -251,3 +251,112 @@ export type VouchersBulkUpdateMetadataResponse = {
 export type VouchersBulkUpdateResponse = {
 	async_action_id: string
 }
+
+export interface ListGiftCardTransactionsQuery {
+	limit?: number
+	page?: number
+}
+
+export interface ListGiftCardTransactionsResponse {
+	object: 'list'
+	data_ref: 'data'
+	data: ListGiftCardTransactionsTypeDataObject[] | []
+	total?: number
+	has_more?: boolean
+}
+
+export type ListGiftCardTransactionsType =
+	| 'CREDITS_REDEMPTION'
+	| 'CREDITS_REFUND'
+	| 'CREDITS_ADDITION'
+	| 'CREDITS_REMOVAL'
+
+export interface ListGiftCardTransactionsTypeDataObject {
+	id: string
+	source_id: string | null
+	voucher_id: string
+	campaign_id: string | null
+	source: 'voucherify-web-ui' | 'API' | null
```

Comment:
are you sure about that?

Reply:
I received a response in which the `source` is `null`
The `id` is always in the response (id of the transaction)
`source_id` is not defined in many cases, but it's returned in every response
`voucher_id` must be defined
if it's a standalone the `campaign_id` could be null

Reply:
or perhaps you are referring to some specific property? :)